judge/utils/problems.py

```diff
@@ -24,8 +24,8 @@ def contest_completed_ids(participation):
     key = 'contest_complete:%d' % participation.id
     result = cache.get(key)
     if result is None:
-        result = set(participation.submissions.filter(submission__result='AC', points=F('problem__points'))
-                     .values_list('problem__problem__id', flat=True).distinct())
+        result = set(participation.submissions.filter(submission__result='AC', points__gte=F('problem__points'))
```

Comment:
Could we perhaps use `case_points` / `case_total` here as well?
```suggestion
        result = set(participation.submissions.filter(submission__result='AC', submission__case_points=F('submission__case_total'))
```

Reply:
This is a ContestSubmission. Is it still ok to check `case_points`/`case_total`?